### ISSUE TITLE
Make queries into an AST node

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#12331]
Today queries are represented as a DefExpr inserted into the AST where it occurs.  It would probably make checking and compiler transformations easier (and less mystical) if we created a QueryExpr type or something that we can check against.

**Comment by dlongnecke-cray** [comment]
What is a query? Because in `compiler/next` when I hear `query` I think of the query framework.

**Comment by dlongnecke-cray** [comment]
I have yet to add a `uAST` node for `?` but yeah maybe we could call it `TypeQuery` or something.

**Comment by mppf** [comment]
Right, we're talking about the `?` thing which the spec calls a query -- https://chapel-lang.org/docs/language/spec/generics.html#formal-arguments-with-queried-types

**Comment by dlongnecke-cray** [comment]
Thanks!!!
